python/cudf/cudf/core/dataframe.py

```diff
@@ -5752,26 +5752,23 @@ def _prepare_for_rowwise_op(self, method, skipna):
             )
             raise ValueError(msg)
 
-        is_pure_dt = all(is_datetime_dtype(dt) for dt in self.dtypes)
-
-        if not is_pure_dt:
+        if numeric_only:
             filtered = self.select_dtypes(include=[np.number, np.bool_])
         else:
             filtered = self.copy(deep=False)
 
-        common_dtype = find_common_type(filtered.dtypes)
+        is_pure_dt = all(is_datetime_dtype(dt) for dt in self.dtypes)
 
-        if filtered._num_columns < self._num_columns:
-            # When we update our pandas compatibility target to 2.0, pandas
-            # will stop supporting numeric_only=None and users will have to
-            # specify True/False. At that time we should also top our implicit
-            # removal of non-numeric columns here.
-            assert Version(pd.__version__) < Version("2.0.0")
-            msg = (
-                "Row-wise operations currently only support int, float "
-                "and bool dtypes. Non numeric columns are ignored."
+        common_dtype = find_common_type(filtered.dtypes)
+        if (
+            not numeric_only
+            and is_string_dtype(common_dtype)
+            and any(not is_string_dtype(dt) for dt in filtered.dtypes)
```

Comment:
Why is `is_string_dtype` the condition here? I get that we can't just verify that every column's dtype is the same as `common_dtype` since we need to allow for casting between different numeric types, but does this condition account for other column dtypes or are those handled separately?

Reply:
The actual intention was to capture `cudf.dtype(object)` dtype, `is_string_dtype` helps capture that. The reason being whenever there seems to be a mix of un-related types, the result is always `object` dtype or an error(incase of mix of struct & list types).
```python
In [3]: find_common_type(["float", "int"])
Out[3]: dtype('float64')

In [4]: find_common_type(["float", "int", "datetime64[ns]"])
Out[4]: dtype('O')
```

Reply:
Ah I see, so basically if we can't compute a common dtype that isn't object we throw an error unless object is actually the result of a bunch of string columns in which case we're OK.